src/OpenTelemetry.Exporter.Geneva/Metrics/GenevaMetricExporter.cs

```diff
@@ -22,26 +23,41 @@ public class GenevaMetricExporter : BaseExporter<Metric>
 
     internal const string DimensionKeyForCustomMetricsNamespace = "_microsoft_metrics_namespace";
 
-    private readonly TlvMetricExporter exporter;
+    private readonly IDisposable exporter;
 
     private delegate ExportResult ExportMetricsFunc(in Batch<Metric> batch);
 
     private readonly ExportMetricsFunc exportMetrics;
 
     private bool isDisposed;
 
+    private Resource resource;
+
+    internal Resource Resource => this.resource ??= this.ParentProvider.GetResource();
+
     public GenevaMetricExporter(GenevaMetricExporterOptions options)
     {
         Guard.ThrowIfNull(options);
         Guard.ThrowIfNullOrWhitespace(options.ConnectionString);
 
-        // TODO: parse connection string to check if otlp protobuf format is enabled.
-        // and then enable either TLV Exporter or Protobuf based exporter.
-        var tlvMetricsExporter = new TlvMetricExporter(options);
+        var connectionStringBuilder = new ConnectionStringBuilder(options.ConnectionString);
 
-        this.exportMetrics = tlvMetricsExporter.Export;
+        if (connectionStringBuilder.PrivatePreviewOtlpProtobufMetricExporter != null && connectionStringBuilder.PrivatePreviewOtlpProtobufMetricExporter.Equals(bool.TrueString, StringComparison.OrdinalIgnoreCase))
+        {
+            var otlpProtobufExporter = new OtlpProtobufMetricExporter(() => { return this.Resource; });
```

Comment:
What is the benefit of passing the resource as delegate here?

If I understand correctly, the Resource should only be utilized for a protobuf-based exporter. However, we have a class-level variable declared for it.

Reply:
Because resource is only available after the exporter is built. Note that `GeneveMetricExporter` would be the one configured with MeterProvider so the resource can only be accessed using the `ParentProvider` of `GenevaMetricExporter`.